### ISSUE TITLE
how to fine tune BlipForImageTextRetrieval?

### ISSUE THREAD

**Issue by ScottishFold007** [huggingface/transformers#21231]
### Feature request

how to fine tune BlipForImageTextRetrieval?
Can you borrow some methods from here to achieve this?
https://github.com/salesforce/LAVIS/blob/main/lavis/models/blip_models/blip_retrieval.py

### Motivation

Implement a graphical matching model that, due to the filtering of poor quality pairs of matches

### Your contribution

Not available at the moment

**Comment by sgugger** [comment]
cc @younesbelkada

**Comment by NielsRogge** [comment]
I'd recommend fine-tuning CLIP if you want to do image-text retrieval using this script: https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text.

Fine-tuning BLIP might be harder as it involves some very specific loss functions.

**Comment by ScottishFold007** [comment]
Thank you for your answer! I tried the fintuning of clip, and it was successful. I use the blip model because I want to use its text matching (binary classification model) to filter out noisy data that is not matched by the text diagram. Because I collect a large number of unlabeled pictures from the Internet, I want to use the blip caption model to tag them, and then filter the invalid image data.

**Comment by NielsRogge** [comment]
CLIP can also be used for image-text matching, by just encoding the image, encoding the text, and computing a cosine similarity score between the respective embeddings.

**Comment by ScottishFold007** [comment]
> CLIP can also be used for image-text matching, by just encoding the image, encoding the text, and computing a cosine similarity score between the respective embeddings.

In fact, what I want to express is that this graphical text matching classifier is similar to the cross encoder in text matching, which can get the interaction information between the two by splicing the graphical embedding, so the accuracy will be higher than the clip (similar to the bi-encoder in text matching)
